core/src/main/scala/cats/Apply.scala

```diff
@@ -39,10 +39,10 @@ trait Apply[F[_]] extends Functor[F] with ApplyArityFunctions[F] { self =>
    * val y = Some(List(10, 20))
    * ap.map2(x, y)(_ + _) == Some(List(11, 12, 21, 22))
    */
-  def compose[G[_]](implicit GG: Apply[G]): Apply[({type λ[α] = F[G[α]]})#λ] =
-    new CompositeApply[F,G] {
-      implicit def F: Apply[F] = self
-      implicit def G: Apply[G] = GG
+  def compose[G[_]: Apply]: Apply[Lambda[X => F[G[X]]]] =
```

Comment:
In #27 it sounds like the agreed upon convention (that we still need to add to the contributor guide) is to use the `implicit GG: Apply[G]` approach instead of using `G[_]: Apply` and then calling `Apply[G]`. Could you please change this back to the previous form? The change from type lambda to `Lambda` is good - keep that change.

Could you also do the same change in the other places in this diff where `<Typeclass>.apply` is being called to fetch the implicit?